sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -841,15 +852,7 @@ class Analyzer(
 
       case q: LogicalPlan =>
         logTrace(s"Attempting to resolve ${q.simpleString}")
-        q.transformExpressionsUp  {
```

Comment:
can you provide more context here? The previous code resolve reference bottom-up, so that we can resolve `a.b.c.d` with one round. The new code changes to top-down, why?